In [1]:
import pandas as pd

circuits = pd.read_csv('data/circuits.csv')
constructor_results = pd.read_csv('data/constructor_results.csv')
constructor_standings = pd.read_csv('data/constructor_standings.csv')
constructors = pd.read_csv('data/constructors.csv')
driver_standings = pd.read_csv('data/driver_standings.csv')
drivers = pd.read_csv('data/drivers.csv')
lap_times = pd.read_csv('data/lap_times.csv')
pit_stops = pd.read_csv('data/pit_stops.csv')
qualifying = pd.read_csv('data/qualifying.csv')
races = pd.read_csv('data/races.csv')
results = pd.read_csv('data/results.csv')
seasons = pd.read_csv('data/seasons.csv')
sprint_results = pd.read_csv('data/sprint_results.csv')
status = pd.read_csv('data/status.csv')

In [2]:
circuits.info()
circuits.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   circuitId   77 non-null     int64  
 1   circuitRef  77 non-null     object 
 2   name        77 non-null     object 
 3   location    77 non-null     object 
 4   country     77 non-null     object 
 5   lat         77 non-null     float64
 6   lng         77 non-null     float64
 7   alt         77 non-null     object 
 8   url         77 non-null     object 
dtypes: float64(2), int64(1), object(6)
memory usage: 5.5+ KB


circuitId        lat         lng
count  77.000000  77.000000   77.000000
mean   39.883117  33.442925    1.076683
std    23.001701  22.808866   65.516951
min     1.000000 -37.849700 -118.189000
25%    20.000000  32.777400   -9.394170
50%    40.000000  40.951700    3.930830
75%    59.000000  46.958900   19.248600
max    80.000000  57.265300  144.968000

In [3]:
constructor_results.info()
constructor_results.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12290 entries, 0 to 12289
Data columns (total 5 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   constructorResultsId  12290 non-null  int64  
 1   raceId                12290 non-null  int64  
 2   constructorId         12290 non-null  int64  
 3   points                12290 non-null  float64
 4   status                12290 non-null  object 
dtypes: float64(1), int64(3), object(1)
memory usage: 480.2+ KB


constructorResultsId        raceId  constructorId        points
count          12290.000000  12290.000000   12290.000000  12290.000000
mean            8191.007242    511.762815      44.661676      3.859967
std             5561.388146    302.677952      57.851043      7.555135
min                1.000000      1.000000       1.000000      0.000000
25%             3073.250000    281.000000       6.000000      0.000000
50%             6145.500000    477.000000      22.000000      0.000000
75%            13698.750000    722.000000      53.000000      4.000000
max            16789.000000   1110.000000     214.000000     66.000000

In [4]:
constructor_standings.info()
constructor_standings.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13051 entries, 0 to 13050
Data columns (total 7 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   constructorStandingsId  13051 non-null  int64  
 1   raceId                  13051 non-null  int64  
 2   constructorId           13051 non-null  int64  
 3   points                  13051 non-null  float64
 4   position                13051 non-null  int64  
 5   positionText            13051 non-null  object 
 6   wins                    13051 non-null  int64  
dtypes: float64(1), int64(5), object(1)
memory usage: 713.9+ KB


constructorStandingsId        raceId  constructorId        points  \
count            13051.000000  13051.000000   13051.000000  13051.000000   
mean             16674.407785    519.974638      48.470232     34.139108   
std               8773.183393    296.269063      59.857491     77.227726   
min                  1.000000      1.000000       1.000000      0.000000   
25%               8798.500000    296.000000       6.000000      0.000000   
50%              20348.000000    497.000000      25.000000      7.000000   
75%              24674.500000    721.000000      57.000000     31.000000   
max              28572.000000   1110.000000     214.000000    765.000000   

           position          wins  
count  13051.000000  13051.000000  
mean       7.271014      0.679182  
std        4.378793      1.823594  
min        1.000000      0.000000  
25%        4.000000      0.000000  
50%        7.000000      0.000000  
75%       10.000000      0.000000  
max       22.000000     19.000000

In [5]:
constructors.info()
constructors.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 211 entries, 0 to 210
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   constructorId   211 non-null    int64 
 1   constructorRef  211 non-null    object
 2   name            211 non-null    object
 3   nationality     211 non-null    object
 4   url             211 non-null    object
dtypes: int64(1), object(4)
memory usage: 8.4+ KB


constructorId
count     211.000000
mean      107.037915
std        61.653629
min         1.000000
25%        54.500000
50%       107.000000
75%       159.500000
max       214.000000

In [6]:
driver_standings.info()
driver_standings.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34124 entries, 0 to 34123
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   driverStandingsId  34124 non-null  int64  
 1   raceId             34124 non-null  int64  
 2   driverId           34124 non-null  int64  
 3   points             34124 non-null  float64
 4   position           34124 non-null  int64  
 5   positionText       34124 non-null  object 
 6   wins               34124 non-null  int64  
dtypes: float64(1), int64(5), object(1)
memory usage: 1.8+ MB


driverStandingsId        raceId      driverId        points  \
count       34124.000000  34124.000000  34124.000000  34124.000000   
mean        42535.650656    572.649396    307.284697     13.489987   
std         21729.601277    284.154765    267.228677     35.599709   
min             1.000000      1.000000      1.000000      0.000000   
25%         19649.750000    347.000000     85.000000      0.000000   
50%         49674.500000    595.000000    221.000000      1.000000   
75%         58946.250000    797.000000    509.000000     10.000000   
max         72187.000000   1110.000000    858.000000    454.000000   

           position          wins  
count  34124.000000  34124.000000  
mean      19.897169      0.269429  
std       16.396017      0.995681  
min        1.000000      0.000000  
25%        8.000000      0.000000  
50%       16.000000      0.000000  
75%       26.000000      0.000000  
max      108.000000     15.000000

In [7]:
drivers.info()
drivers.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 857 entries, 0 to 856
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   driverId     857 non-null    int64 
 1   driverRef    857 non-null    object
 2   number       857 non-null    object
 3   code         857 non-null    object
 4   forename     857 non-null    object
 5   surname      857 non-null    object
 6   dob          857 non-null    object
 7   nationality  857 non-null    object
 8   url          857 non-null    object
dtypes: int64(1), object(8)
memory usage: 60.4+ KB


driverId
count  857.000000
mean   429.057176
std    247.632402
min      1.000000
25%    215.000000
50%    429.000000
75%    643.000000
max    858.000000

In [8]:
lap_times.info()
lap_times.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 551742 entries, 0 to 551741
Data columns (total 6 columns):
 #   Column        Non-Null Count   Dtype 
---  ------        --------------   ----- 
 0   raceId        551742 non-null  int64 
 1   driverId      551742 non-null  int64 
 2   lap           551742 non-null  int64 
 3   position      551742 non-null  int64 
 4   time          551742 non-null  object
 5   milliseconds  551742 non-null  int64 
dtypes: int64(5), object(1)
memory usage: 25.3+ MB


raceId       driverId            lap       position  \
count  551742.000000  551742.000000  551742.000000  551742.000000   
mean      564.878467     296.918785      29.990193       9.658699   
std       425.883069     377.935897      18.420399       5.538011   
min         1.000000       1.000000       1.000000       1.000000   
25%       131.000000      15.000000      14.000000       5.000000   
50%       846.000000      39.000000      29.000000       9.000000   
75%       976.000000     817.000000      44.000000      14.000000   
max      1110.000000     858.000000      87.000000      24.000000   

       milliseconds  
count  5.517420e+05  
mean   9.570621e+04  
std    7.478605e+04  
min    5.540400e+04  
25%    8.199300e+04  
50%    9.067300e+04  
75%    1.022430e+05  
max    7.507547e+06

In [9]:
pit_stops.info()
pit_stops.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10089 entries, 0 to 10088
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   raceId        10089 non-null  int64 
 1   driverId      10089 non-null  int64 
 2   stop          10089 non-null  int64 
 3   lap           10089 non-null  int64 
 4   time          10089 non-null  object
 5   duration      10089 non-null  object
 6   milliseconds  10089 non-null  int64 
dtypes: int64(5), object(2)
memory usage: 551.9+ KB


raceId      driverId          stop           lap  milliseconds
count  10089.000000  10089.000000  10089.000000  10089.000000  1.008900e+04
mean     962.774011    523.998910      1.759738     25.312023  7.534863e+04
std       81.144375    389.698555      0.916282     14.729775  2.788588e+05
min      841.000000      1.000000      1.000000      1.000000  1.289700e+04
25%      888.000000     18.000000      1.000000     13.000000  2.191400e+04
50%      958.000000    815.000000      2.000000     25.000000  2.357000e+04
75%     1035.000000    831.000000      2.000000     36.000000  2.620200e+04
max     1110.000000    858.000000      6.000000     78.000000  3.069017e+06

In [10]:
circuits.info()
circuits.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   circuitId   77 non-null     int64  
 1   circuitRef  77 non-null     object 
 2   name        77 non-null     object 
 3   location    77 non-null     object 
 4   country     77 non-null     object 
 5   lat         77 non-null     float64
 6   lng         77 non-null     float64
 7   alt         77 non-null     object 
 8   url         77 non-null     object 
dtypes: float64(2), int64(1), object(6)
memory usage: 5.5+ KB


circuitId        lat         lng
count  77.000000  77.000000   77.000000
mean   39.883117  33.442925    1.076683
std    23.001701  22.808866   65.516951
min     1.000000 -37.849700 -118.189000
25%    20.000000  32.777400   -9.394170
50%    40.000000  40.951700    3.930830
75%    59.000000  46.958900   19.248600
max    80.000000  57.265300  144.968000

In [11]:
qualifying.info()
qualifying.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9815 entries, 0 to 9814
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   qualifyId      9815 non-null   int64 
 1   raceId         9815 non-null   int64 
 2   driverId       9815 non-null   int64 
 3   constructorId  9815 non-null   int64 
 4   number         9815 non-null   int64 
 5   position       9815 non-null   int64 
 6   q1             9807 non-null   object
 7   q2             9651 non-null   object
 8   q3             9488 non-null   object
dtypes: int64(6), object(3)
memory usage: 690.2+ KB


qualifyId       raceId     driverId  constructorId       number  \
count  9815.000000  9815.000000  9815.000000    9815.000000  9815.000000   
mean   4920.076108   589.809985   314.514519      44.783597    18.114519   
std    2847.637640   421.207023   381.264736      70.843628    17.859565   
min       1.000000     1.000000     1.000000       1.000000     0.000000   
25%    2455.500000   119.500000    16.000000       4.000000     7.000000   
50%    4910.000000   856.000000    50.000000       9.000000    14.000000   
75%    7383.500000   981.000000   818.000000      28.000000    22.000000   
max    9872.000000  1110.000000   858.000000     214.000000    99.000000   

          position  
count  9815.000000  
mean     11.244931  
std       6.290625  
min       1.000000  
25%       6.000000  
50%      11.000000  
75%      17.000000  
max      28.000000

In [12]:
races.info()
races.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1101 entries, 0 to 1100
Data columns (total 18 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   raceId       1101 non-null   int64 
 1   year         1101 non-null   int64 
 2   round        1101 non-null   int64 
 3   circuitId    1101 non-null   int64 
 4   name         1101 non-null   object
 5   date         1101 non-null   object
 6   time         1101 non-null   object
 7   url          1101 non-null   object
 8   fp1_date     1101 non-null   object
 9   fp1_time     1101 non-null   object
 10  fp2_date     1101 non-null   object
 11  fp2_time     1101 non-null   object
 12  fp3_date     1101 non-null   object
 13  fp3_time     1101 non-null   object
 14  quali_date   1101 non-null   object
 15  quali_time   1101 non-null   object
 16  sprint_date  1101 non-null   object
 17  sprint_time  1101 non-null   object
dtypes: int64(4), object(14)
memory usage: 155.0+ KB


raceId         year        round    circuitId
count  1101.000000  1101.000000  1101.000000  1101.000000
mean    553.355132  1992.020890     8.494096    23.700272
std     321.425790    20.296406     5.081089    19.346014
min       1.000000  1950.000000     1.000000     1.000000
25%     276.000000  1976.000000     4.000000     9.000000
50%     551.000000  1994.000000     8.000000    18.000000
75%     826.000000  2010.000000    12.000000    34.000000
max    1120.000000  2023.000000    22.000000    80.000000

In [13]:
results.info()
results.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26080 entries, 0 to 26079
Data columns (total 18 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   resultId         26080 non-null  int64  
 1   raceId           26080 non-null  int64  
 2   driverId         26080 non-null  int64  
 3   constructorId    26080 non-null  int64  
 4   number           26080 non-null  object 
 5   grid             26080 non-null  int64  
 6   position         26080 non-null  object 
 7   positionText     26080 non-null  object 
 8   positionOrder    26080 non-null  int64  
 9   points           26080 non-null  float64
 10  laps             26080 non-null  int64  
 11  time             26080 non-null  object 
 12  milliseconds     26080 non-null  object 
 13  fastestLap       26080 non-null  object 
 14  rank             26080 non-null  object 
 15  fastestLapTime   26080 non-null  object 
 16  fastestLapSpeed  26080 non-null  object 
 17  statusId    

resultId        raceId      driverId  constructorId          grid  \
count  26080.000000  26080.000000  26080.000000   26080.000000  26080.000000   
mean   13041.372661    536.695667    266.277569      49.059663     11.167561   
std     7530.008377    303.034639    272.581622      60.221056      7.232797   
min        1.000000      1.000000      1.000000       1.000000      0.000000   
25%     6520.750000    294.750000     57.000000       6.000000      5.000000   
50%    13040.500000    519.000000    163.000000      25.000000     11.000000   
75%    19560.250000    791.000000    364.000000      58.250000     17.000000   
max    26085.000000   1110.000000    858.000000     214.000000     34.000000   

       positionOrder        points          laps      statusId  
count   26080.000000  26080.000000  26080.000000  26080.000000  
mean       12.854141      1.906635     46.076687     17.476074  
std         7.700068      4.219715     29.726058     26.129965  
min         1.000000      0.000000      0.000000      1.000000  
25%         6.000000      0.000000     22.000000      1.000000  
50%        12.000000      0.000000     53.000000     10.000000  
75%        18.000000      2.000000     66.000000     14.000000  
max        39.000000     50.000000    200.000000    141.000000

In [14]:
seasons.info()
seasons.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 74 entries, 0 to 73
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   year    74 non-null     int64 
 1   url     74 non-null     object
dtypes: int64(1), object(1)
memory usage: 1.3+ KB


year
count    74.000000
mean   1986.500000
std      21.505813
min    1950.000000
25%    1968.250000
50%    1986.500000
75%    2004.750000
max    2023.000000

In [15]:
sprint_results.info()
sprint_results.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180 entries, 0 to 179
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   resultId        180 non-null    int64 
 1   raceId          180 non-null    int64 
 2   driverId        180 non-null    int64 
 3   constructorId   180 non-null    int64 
 4   number          180 non-null    int64 
 5   grid            180 non-null    int64 
 6   position        180 non-null    object
 7   positionText    180 non-null    object
 8   positionOrder   180 non-null    int64 
 9   points          180 non-null    int64 
 10  laps            180 non-null    int64 
 11  time            180 non-null    object
 12  milliseconds    180 non-null    object
 13  fastestLap      180 non-null    object
 14  fastestLapTime  180 non-null    object
 15  statusId        180 non-null    int64 
dtypes: int64(10), object(6)
memory usage: 22.6+ KB


resultId       raceId    driverId  constructorId      number  \
count  180.000000   180.000000  180.000000     180.000000  180.000000   
mean    90.500000  1085.666667  714.088889      95.500000   27.550000   
std     52.105662    17.362388  298.112841      88.359632   24.273874   
min      1.000000  1061.000000    1.000000       1.000000    1.000000   
25%     45.750000  1071.000000  817.000000       6.000000   10.000000   
50%     90.500000  1084.000000  840.000000      84.000000   20.000000   
75%    135.250000  1101.000000  847.000000     210.000000   44.000000   
max    180.000000  1110.000000  858.000000     214.000000   99.000000   

             grid  positionOrder      points        laps    statusId  
count  180.000000     180.000000  180.000000  180.000000  180.000000  
mean    10.100000      10.500000    1.300000   19.216667    3.983333  
std      5.861626       5.782366    2.318013    5.469341   17.662091  
min      0.000000       1.000000    0.000000    0.000000    1.000000  
25%      5.000000       5.750000    0.000000   17.000000    1.000000  
50%     10.000000      10.500000    0.000000   21.000000    1.000000  
75%     15.000000      15.250000    2.000000   24.000000    1.000000  
max     20.000000      20.000000    8.000000   24.000000  130.000000

In [16]:
status.info()
status.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 139 entries, 0 to 138
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   statusId  139 non-null    int64 
 1   status    139 non-null    object
dtypes: int64(1), object(1)
memory usage: 2.3+ KB


statusId
count  139.000000
mean    71.237410
std     41.092434
min      1.000000
25%     35.500000
50%     72.000000
75%    106.500000
max    141.000000

In [24]:
def custom_info(dataframe, dataframe_name="Unnamed DataFrame"):
    print(f"\nChecking for \\N values in {dataframe_name}:")

    info_df = pd.DataFrame(index=dataframe.columns, columns=['Non-Null Count', 'Null Count', 'Unique Values', 'Data Type'])

    for column in dataframe.columns:
        non_null_count = dataframe[column].count()
        null_count = dataframe[column].eq('\\N').sum()
        unique_values = dataframe[column].nunique()
        data_type = dataframe[column].dtype

        info_df.loc[column] = [non_null_count, null_count, unique_values, data_type]

        # Print column-specific information if \N values are found
        if null_count > 0:
            print(f"\\N values found in column '{column}'")

    print(info_df)

# List of DataFrame variable names and corresponding names
dataframes_and_names = [
    (circuits, 'circuits'),
    (constructor_results, 'constructor_results'),
    (constructor_standings, 'constructor_standings'),
    (constructors, 'constructors'),
    (driver_standings, 'driver_standings'),
    (drivers, 'drivers'),
    (lap_times, 'lap_times'),
    (pit_stops, 'pit_stops'),
    (qualifying, 'qualifying'),
    (races, 'races'),
    (results, 'results'),
    (seasons, 'seasons'),
    (sprint_results, 'sprint_results'),
    (status, 'status')
]

# Apply custom_info to each DataFrame
for df, name in dataframes_and_names:
    custom_info(df, dataframe_name=name)



Checking for \N values in circuits:
\N values found in column 'alt'
           Non-Null Count Null Count Unique Values Data Type
circuitId              77          0            77     int64
circuitRef             77          0            77    object
name                   77          0            77    object
location               77          0            75    object
country                77          0            35    object
lat                    77          0            77   float64
lng                    77          0            77   float64
alt                    77          3            66    object
url                    77          0            77    object

Checking for \N values in constructor_results:
\N values found in column 'status'
                     Non-Null Count Null Count Unique Values Data Type
constructorResultsId          12290          0         12290     int64
raceId                        12290          0          1028     int64
constructorId            

import pandas as pd

​

circuits = pd.read_csv('data/circuits.csv')

constructor_results = pd.read_csv('data/constructor_results.csv')

constructor_standings = pd.read_csv('data/constructor_standings.csv')

constructors = pd.read_csv('data/constructors.csv')

driver_standings = pd.read_csv('data/driver_standings.csv')

drivers = pd.read_csv('data/drivers.csv')

lap_times = pd.read_csv('data/lap_times.csv')

pit_stops = pd.read_csv('data/pit_stops.csv')

qualifying = pd.read_csv('data/qualifying.csv')

races = pd.read_csv('data/races.csv')

results = pd.read_csv('data/results.csv')

seasons = pd.read_csv('data/seasons.csv')

sprint_results = pd.read_csv('data/sprint_results.csv')

status = pd.read_csv('data/status.csv')

circuits.info()

circuits.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   circuitId   77 non-null     int64  
 1   circuitRef  77 non-null     object 
 2   name        77 non-null     object 
 3   location    77 non-null     object 
 4   country     77 non-null     object 
 5   lat         77 non-null     float64
 6   lng         77 non-null     float64
 7   alt         77 non-null     object 
 8   url         77 non-null     object 
dtypes: float64(2), int64(1), object(6)
memory usage: 5.5+ KB

	circuitId 	lat 	lng
count 	77.000000 	77.000000 	77.000000
mean 	39.883117 	33.442925 	1.076683
std 	23.001701 	22.808866 	65.516951
min 	1.000000 	-37.849700 	-118.189000
25% 	20.000000 	32.777400 	-9.394170
50% 	40.000000 	40.951700 	3.930830
75% 	59.000000 	46.958900 	19.248600
max 	80.000000 	57.265300 	144.968000

constructor_results.info()

constructor_results.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12290 entries, 0 to 12289
Data columns (total 5 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   constructorResultsId  12290 non-null  int64  
 1   raceId                12290 non-null  int64  
 2   constructorId         12290 non-null  int64  
 3   points                12290 non-null  float64
 4   status                12290 non-null  object 
dtypes: float64(1), int64(3), object(1)
memory usage: 480.2+ KB

	constructorResultsId 	raceId 	constructorId 	points
count 	12290.000000 	12290.000000 	12290.000000 	12290.000000
mean 	8191.007242 	511.762815 	44.661676 	3.859967
std 	5561.388146 	302.677952 	57.851043 	7.555135
min 	1.000000 	1.000000 	1.000000 	0.000000
25% 	3073.250000 	281.000000 	6.000000 	0.000000
50% 	6145.500000 	477.000000 	22.000000 	0.000000
75% 	13698.750000 	722.000000 	53.000000 	4.000000
max 	16789.000000 	1110.000000 	214.000000 	66.000000

constructor_standings.info()

constructor_standings.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13051 entries, 0 to 13050
Data columns (total 7 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   constructorStandingsId  13051 non-null  int64  
 1   raceId                  13051 non-null  int64  
 2   constructorId           13051 non-null  int64  
 3   points                  13051 non-null  float64
 4   position                13051 non-null  int64  
 5   positionText            13051 non-null  object 
 6   wins                    13051 non-null  int64  
dtypes: float64(1), int64(5), object(1)
memory usage: 713.9+ KB

	constructorStandingsId 	raceId 	constructorId 	points 	position 	wins
count 	13051.000000 	13051.000000 	13051.000000 	13051.000000 	13051.000000 	13051.000000
mean 	16674.407785 	519.974638 	48.470232 	34.139108 	7.271014 	0.679182
std 	8773.183393 	296.269063 	59.857491 	77.227726 	4.378793 	1.823594
min 	1.000000 	1.000000 	1.000000 	0.000000 	1.000000 	0.000000
25% 	8798.500000 	296.000000 	6.000000 	0.000000 	4.000000 	0.000000
50% 	20348.000000 	497.000000 	25.000000 	7.000000 	7.000000 	0.000000
75% 	24674.500000 	721.000000 	57.000000 	31.000000 	10.000000 	0.000000
max 	28572.000000 	1110.000000 	214.000000 	765.000000 	22.000000 	19.000000

constructors.info()

constructors.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 211 entries, 0 to 210
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   constructorId   211 non-null    int64 
 1   constructorRef  211 non-null    object
 2   name            211 non-null    object
 3   nationality     211 non-null    object
 4   url             211 non-null    object
dtypes: int64(1), object(4)
memory usage: 8.4+ KB

	constructorId
count 	211.000000
mean 	107.037915
std 	61.653629
min 	1.000000
25% 	54.500000
50% 	107.000000
75% 	159.500000
max 	214.000000

driver_standings.info()

driver_standings.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34124 entries, 0 to 34123
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   driverStandingsId  34124 non-null  int64  
 1   raceId             34124 non-null  int64  
 2   driverId           34124 non-null  int64  
 3   points             34124 non-null  float64
 4   position           34124 non-null  int64  
 5   positionText       34124 non-null  object 
 6   wins               34124 non-null  int64  
dtypes: float64(1), int64(5), object(1)
memory usage: 1.8+ MB

	driverStandingsId 	raceId 	driverId 	points 	position 	wins
count 	34124.000000 	34124.000000 	34124.000000 	34124.000000 	34124.000000 	34124.000000
mean 	42535.650656 	572.649396 	307.284697 	13.489987 	19.897169 	0.269429
std 	21729.601277 	284.154765 	267.228677 	35.599709 	16.396017 	0.995681
min 	1.000000 	1.000000 	1.000000 	0.000000 	1.000000 	0.000000
25% 	19649.750000 	347.000000 	85.000000 	0.000000 	8.000000 	0.000000
50% 	49674.500000 	595.000000 	221.000000 	1.000000 	16.000000 	0.000000
75% 	58946.250000 	797.000000 	509.000000 	10.000000 	26.000000 	0.000000
max 	72187.000000 	1110.000000 	858.000000 	454.000000 	108.000000 	15.000000

drivers.info()

drivers.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 857 entries, 0 to 856
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   driverId     857 non-null    int64 
 1   driverRef    857 non-null    object
 2   number       857 non-null    object
 3   code         857 non-null    object
 4   forename     857 non-null    object
 5   surname      857 non-null    object
 6   dob          857 non-null    object
 7   nationality  857 non-null    object
 8   url          857 non-null    object
dtypes: int64(1), object(8)
memory usage: 60.4+ KB

	driverId
count 	857.000000
mean 	429.057176
std 	247.632402
min 	1.000000
25% 	215.000000
50% 	429.000000
75% 	643.000000
max 	858.000000

lap_times.info()

lap_times.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 551742 entries, 0 to 551741
Data columns (total 6 columns):
 #   Column        Non-Null Count   Dtype 
---  ------        --------------   ----- 
 0   raceId        551742 non-null  int64 
 1   driverId      551742 non-null  int64 
 2   lap           551742 non-null  int64 
 3   position      551742 non-null  int64 
 4   time          551742 non-null  object
 5   milliseconds  551742 non-null  int64 
dtypes: int64(5), object(1)
memory usage: 25.3+ MB

	raceId 	driverId 	lap 	position 	milliseconds
count 	551742.000000 	551742.000000 	551742.000000 	551742.000000 	5.517420e+05
mean 	564.878467 	296.918785 	29.990193 	9.658699 	9.570621e+04
std 	425.883069 	377.935897 	18.420399 	5.538011 	7.478605e+04
min 	1.000000 	1.000000 	1.000000 	1.000000 	5.540400e+04
25% 	131.000000 	15.000000 	14.000000 	5.000000 	8.199300e+04
50% 	846.000000 	39.000000 	29.000000 	9.000000 	9.067300e+04
75% 	976.000000 	817.000000 	44.000000 	14.000000 	1.022430e+05
max 	1110.000000 	858.000000 	87.000000 	24.000000 	7.507547e+06

pit_stops.info()

pit_stops.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10089 entries, 0 to 10088
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   raceId        10089 non-null  int64 
 1   driverId      10089 non-null  int64 
 2   stop          10089 non-null  int64 
 3   lap           10089 non-null  int64 
 4   time          10089 non-null  object
 5   duration      10089 non-null  object
 6   milliseconds  10089 non-null  int64 
dtypes: int64(5), object(2)
memory usage: 551.9+ KB

	raceId 	driverId 	stop 	lap 	milliseconds
count 	10089.000000 	10089.000000 	10089.000000 	10089.000000 	1.008900e+04
mean 	962.774011 	523.998910 	1.759738 	25.312023 	7.534863e+04
std 	81.144375 	389.698555 	0.916282 	14.729775 	2.788588e+05
min 	841.000000 	1.000000 	1.000000 	1.000000 	1.289700e+04
25% 	888.000000 	18.000000 	1.000000 	13.000000 	2.191400e+04
50% 	958.000000 	815.000000 	2.000000 	25.000000 	2.357000e+04
75% 	1035.000000 	831.000000 	2.000000 	36.000000 	2.620200e+04
max 	1110.000000 	858.000000 	6.000000 	78.000000 	3.069017e+06

circuits.info()

circuits.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   circuitId   77 non-null     int64  
 1   circuitRef  77 non-null     object 
 2   name        77 non-null     object 
 3   location    77 non-null     object 
 4   country     77 non-null     object 
 5   lat         77 non-null     float64
 6   lng         77 non-null     float64
 7   alt         77 non-null     object 
 8   url         77 non-null     object 
dtypes: float64(2), int64(1), object(6)
memory usage: 5.5+ KB

	circuitId 	lat 	lng
count 	77.000000 	77.000000 	77.000000
mean 	39.883117 	33.442925 	1.076683
std 	23.001701 	22.808866 	65.516951
min 	1.000000 	-37.849700 	-118.189000
25% 	20.000000 	32.777400 	-9.394170
50% 	40.000000 	40.951700 	3.930830
75% 	59.000000 	46.958900 	19.248600
max 	80.000000 	57.265300 	144.968000

qualifying.info()

qualifying.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9815 entries, 0 to 9814
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   qualifyId      9815 non-null   int64 
 1   raceId         9815 non-null   int64 
 2   driverId       9815 non-null   int64 
 3   constructorId  9815 non-null   int64 
 4   number         9815 non-null   int64 
 5   position       9815 non-null   int64 
 6   q1             9807 non-null   object
 7   q2             9651 non-null   object
 8   q3             9488 non-null   object
dtypes: int64(6), object(3)
memory usage: 690.2+ KB

	qualifyId 	raceId 	driverId 	constructorId 	number 	position
count 	9815.000000 	9815.000000 	9815.000000 	9815.000000 	9815.000000 	9815.000000
mean 	4920.076108 	589.809985 	314.514519 	44.783597 	18.114519 	11.244931
std 	2847.637640 	421.207023 	381.264736 	70.843628 	17.859565 	6.290625
min 	1.000000 	1.000000 	1.000000 	1.000000 	0.000000 	1.000000
25% 	2455.500000 	119.500000 	16.000000 	4.000000 	7.000000 	6.000000
50% 	4910.000000 	856.000000 	50.000000 	9.000000 	14.000000 	11.000000
75% 	7383.500000 	981.000000 	818.000000 	28.000000 	22.000000 	17.000000
max 	9872.000000 	1110.000000 	858.000000 	214.000000 	99.000000 	28.000000

races.info()

races.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1101 entries, 0 to 1100
Data columns (total 18 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   raceId       1101 non-null   int64 
 1   year         1101 non-null   int64 
 2   round        1101 non-null   int64 
 3   circuitId    1101 non-null   int64 
 4   name         1101 non-null   object
 5   date         1101 non-null   object
 6   time         1101 non-null   object
 7   url          1101 non-null   object
 8   fp1_date     1101 non-null   object
 9   fp1_time     1101 non-null   object
 10  fp2_date     1101 non-null   object
 11  fp2_time     1101 non-null   object
 12  fp3_date     1101 non-null   object
 13  fp3_time     1101 non-null   object
 14  quali_date   1101 non-null   object
 15  quali_time   1101 non-null   object
 16  sprint_date  1101 non-null   object
 17  sprint_time  1101 non-null   object
dtypes: int64(4), object(14)
memory usage: 155.0+ KB

	raceId 	year 	round 	circuitId
count 	1101.000000 	1101.000000 	1101.000000 	1101.000000
mean 	553.355132 	1992.020890 	8.494096 	23.700272
std 	321.425790 	20.296406 	5.081089 	19.346014
min 	1.000000 	1950.000000 	1.000000 	1.000000
25% 	276.000000 	1976.000000 	4.000000 	9.000000
50% 	551.000000 	1994.000000 	8.000000 	18.000000
75% 	826.000000 	2010.000000 	12.000000 	34.000000
max 	1120.000000 	2023.000000 	22.000000 	80.000000

results.info()

results.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26080 entries, 0 to 26079
Data columns (total 18 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   resultId         26080 non-null  int64  
 1   raceId           26080 non-null  int64  
 2   driverId         26080 non-null  int64  
 3   constructorId    26080 non-null  int64  
 4   number           26080 non-null  object 
 5   grid             26080 non-null  int64  
 6   position         26080 non-null  object 
 7   positionText     26080 non-null  object 
 8   positionOrder    26080 non-null  int64  
 9   points           26080 non-null  float64
 10  laps             26080 non-null  int64  
 11  time             26080 non-null  object 
 12  milliseconds     26080 non-null  object 
 13  fastestLap       26080 non-null  object 
 14  rank             26080 non-null  object 
 15  fastestLapTime   26080 non-null  object 
 16  fastestLapSpeed  26080 non-null  object 
 17  statusId         26080 non-null  int64  
dtypes: float64(1), int64(8), object(9)
memory usage: 3.6+ MB

	resultId 	raceId 	driverId 	constructorId 	grid 	positionOrder 	points 	laps 	statusId
count 	26080.000000 	26080.000000 	26080.000000 	26080.000000 	26080.000000 	26080.000000 	26080.000000 	26080.000000 	26080.000000
mean 	13041.372661 	536.695667 	266.277569 	49.059663 	11.167561 	12.854141 	1.906635 	46.076687 	17.476074
std 	7530.008377 	303.034639 	272.581622 	60.221056 	7.232797 	7.700068 	4.219715 	29.726058 	26.129965
min 	1.000000 	1.000000 	1.000000 	1.000000 	0.000000 	1.000000 	0.000000 	0.000000 	1.000000
25% 	6520.750000 	294.750000 	57.000000 	6.000000 	5.000000 	6.000000 	0.000000 	22.000000 	1.000000
50% 	13040.500000 	519.000000 	163.000000 	25.000000 	11.000000 	12.000000 	0.000000 	53.000000 	10.000000
75% 	19560.250000 	791.000000 	364.000000 	58.250000 	17.000000 	18.000000 	2.000000 	66.000000 	14.000000
max 	26085.000000 	1110.000000 	858.000000 	214.000000 	34.000000 	39.000000 	50.000000 	200.000000 	141.000000

seasons.info()

seasons.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 74 entries, 0 to 73
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   year    74 non-null     int64 
 1   url     74 non-null     object
dtypes: int64(1), object(1)
memory usage: 1.3+ KB

	year
count 	74.000000
mean 	1986.500000
std 	21.505813
min 	1950.000000
25% 	1968.250000
50% 	1986.500000
75% 	2004.750000
max 	2023.000000

sprint_results.info()

sprint_results.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180 entries, 0 to 179
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   resultId        180 non-null    int64 
 1   raceId          180 non-null    int64 
 2   driverId        180 non-null    int64 
 3   constructorId   180 non-null    int64 
 4   number          180 non-null    int64 
 5   grid            180 non-null    int64 
 6   position        180 non-null    object
 7   positionText    180 non-null    object
 8   positionOrder   180 non-null    int64 
 9   points          180 non-null    int64 
 10  laps            180 non-null    int64 
 11  time            180 non-null    object
 12  milliseconds    180 non-null    object
 13  fastestLap      180 non-null    object
 14  fastestLapTime  180 non-null    object
 15  statusId        180 non-null    int64 
dtypes: int64(10), object(6)
memory usage: 22.6+ KB

	resultId 	raceId 	driverId 	constructorId 	number 	grid 	positionOrder 	points 	laps 	statusId
count 	180.000000 	180.000000 	180.000000 	180.000000 	180.000000 	180.000000 	180.000000 	180.000000 	180.000000 	180.000000
mean 	90.500000 	1085.666667 	714.088889 	95.500000 	27.550000 	10.100000 	10.500000 	1.300000 	19.216667 	3.983333
std 	52.105662 	17.362388 	298.112841 	88.359632 	24.273874 	5.861626 	5.782366 	2.318013 	5.469341 	17.662091
min 	1.000000 	1061.000000 	1.000000 	1.000000 	1.000000 	0.000000 	1.000000 	0.000000 	0.000000 	1.000000
25% 	45.750000 	1071.000000 	817.000000 	6.000000 	10.000000 	5.000000 	5.750000 	0.000000 	17.000000 	1.000000
50% 	90.500000 	1084.000000 	840.000000 	84.000000 	20.000000 	10.000000 	10.500000 	0.000000 	21.000000 	1.000000
75% 	135.250000 	1101.000000 	847.000000 	210.000000 	44.000000 	15.000000 	15.250000 	2.000000 	24.000000 	1.000000
max 	180.000000 	1110.000000 	858.000000 	214.000000 	99.000000 	20.000000 	20.000000 	8.000000 	24.000000 	130.000000

status.info()

status.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 139 entries, 0 to 138
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   statusId  139 non-null    int64 
 1   status    139 non-null    object
dtypes: int64(1), object(1)
memory usage: 2.3+ KB

	statusId
count 	139.000000
mean 	71.237410
std 	41.092434
min 	1.000000
25% 	35.500000
50% 	72.000000
75% 	106.500000
max 	141.000000